packages/happy-dom/src/xml-http-request/XMLHttpRequest.ts

```diff
@@ -0,0 +1,654 @@
+			case 'file:':
+				return this.sendFile(data);
+			case 'http:':
+			case 'https:':
+				return this.sendHttp(data);
+			default:
+				throw new DOMException(`Unsupported protocol ${this.url.protocol}`, networkError);
+		}
+	}
+
+	/**
+	 * Aborts the request if it has already been sent.
+	 */
+	public abort(): void {
+		if (this._request == null) {
+			return;
+		}
+		// ClientRequest.destroy breaks the test suite for versions 10 and 12,
+		// Hence the version check
+		if (MajorNodeVersion > 13) {
+			this._request.destroy();
+		} else {
+			this._request.abort();
+		}
+		this.setReadyState(XMLHttpRequest.DONE);
+		this.setError();
+
+		this.dispatchProgress('abort');
+		this.dispatchProgress('loadend');
+	}
+
+	/**
+	 * Returns the string containing the text of the specified header, or null if either the response has not yet been received or the header doesn't exist in the response.
+	 *
+	 * @param name The name of the header.
+	 */
+	public getResponseHeader(name: string): string {
+		if (
+			this.responseHeaders == null ||
+			name == null ||
+			this.readyState in [XMLHttpRequest.OPENED, XMLHttpRequest.UNSENT]
+		) {
+			return null;
+		}
+		const loweredName = name.toLowerCase();
+		return this.responseHeaders.hasOwnProperty(loweredName)
+			? this.responseHeaders[name.toLowerCase()]
+			: null;
+	}
+
+	/**
+	 * Returns all the response headers, separated by CRLF, as a string, or null if no response has been received.
+	 *
+	 */
+	public getAllResponseHeaders(): string {
+		if (
+			this.responseHeaders == null ||
+			this.readyState in [XMLHttpRequest.OPENED, XMLHttpRequest.UNSENT]
+		) {
+			return '';
+		}
+		return Object.keys(this.responseHeaders)
+			.map((key) => `${key}: ${this.responseHeaders[key]}`)
+			.join('\r\n');
+	}
+
+	/**
+	 * Overrides the MIME type returned by the server.
+	 *
+	 * @param mimeType The MIME type to use.
+	 */
+	public overrideMimeType(mimeType: string): void {
+		if (this.readyState in [XMLHttpRequest.LOADING, XMLHttpRequest.DONE]) {
+			throw new DOMException(
+				'overrideMimeType() not allowed in LOADING or DONE',
+				DOMExceptionNameEnum.invalidStateError
+			);
+		}
+		this.mimeOverride = mimeType.toLowerCase();
+	}
+
+	/**
+	 * Sets the value of the ReadyState.
+	 *
+	 * @param readyState The new value of the ReadyState.
+	 */
+	private setReadyState(readyState: number): void {
+		this.readyState = readyState;
+		this.dispatchEvent(new ProgressEvent('readystatechange'));
+	}
+
+	/**
+	 * Send request with file.
+	 *
+	 * @todo Not implemented.
+	 * @param _data File body to send.
+	 */
+	private sendFile(_data: unknown): void {
+		// TODO: sendFile() not implemented.
+		throw new Error('Protocol file: not implemented');
+	}
+
+	/**
+	 * Send request with http.
+	 *
+	 * @param data Data to send.
+	 */
+	private sendHttp(data?: string | Buffer | ArrayBuffer | ArrayBufferView): void {
+		const { _defaultView } = XMLHttpRequest;
+		if (this.sync) {
+			// TODO: sync not implemented.
```

Comment:
Need to implement it. A possible solution: use spawnSync run other process to realize the function.
Refer in https://github.com/jsdom/jsdom/tree/master/lib/jsdom/living/xhr

Reply:
I have a lot of problems with how to pass and call the parameters of the parent and child processes.🥲